core/services/assets/Registry.php

```diff
@@ -256,6 +283,57 @@ public function getData($key)
     }
 
 
+    /**
+     * Get the actual asset path for asset manifests.
+     * If there is no asset path found for the given $chunk_name, then the $chunk_name is returned.
+     * @param string $chunk_name
+     * @param string $asset_type
+     * @return string
+     * @throws InvalidFilePathException
+     * @since $VID:$
+     */
+    public function getAssetUrl($chunk_name, $asset_type)
+    {
+        if (empty($this->cached_manifests)) {
+            $this->registerManifests();
+        }
+        return isset($this->cached_manifests[$chunk_name][$asset_type])
+            ? $this->domain->distributionAssetsUrl() . $this->cached_manifests[$chunk_name][$asset_type]
+            : $chunk_name;
+    }
+
+
+    /**
+     * Register any asset manifests.
+     * @throws InvalidFilePathException
+     * @since $VID:$
+     */
+    private function registerManifests()
+    {
+        if (! empty($this->cached_manifests)) {
+            //already registered get out.  This usually means that a $chunk_name (argument for getAssetUrl call) either
+            //doesn't exist in any registered manifest files.  This could be because the $chunk_name is an actual file
+            //url and thus not registered in the manifest, or a plugin registered its manifest files too late.
+            return;
+        }
+        $registered_manifest_files = apply_filters(
+            'FHEE__EventEspresso_core_services_assets_Registry__registerManifests__registered_manifest_files',
+            $this->domain->distributionAssetsPath() . 'build-manifest.json'
```

Comment:
I added this filter because this will allow our add-ons to register their manifests to automatically be cached as well (once add-ons start using the same build process).  The only catch to this is add-ons will have to make sure they register their manifests early enough in the loading process so we may want to add an official Plugin registration API interface for this (and in turn add it to the `EE_Register_Addon` api?

Reply:
ya agreed, however I wonder if we should still keep the filter for plugin authors who want to do things the "wp" way vs going through our plugin api if they have a really simple extension.  Or would you rather we enforce registering via our plugin api?